src/compile/selection/selection.ts

```diff
@@ -202,6 +212,32 @@ export function assembleLayerSelectionMarks(model: LayerModel, marks: any[]): an
   return clipGroup ? clipMarks(marks) : marks;
 }
 
+const IDENTIFIER_TRANSFORM: VgIdentifierTransform = {type: 'identifier', as: SELECTION_ID};
+export function addIdentifierTransforms(model: Model, data: VgData[]) {
+  let identifier = false;
+  forEachSelection(model, (selCmpt) => {
+    identifier = identifier || selCmpt.project.some((proj) => proj.field === SELECTION_ID);
+  });
+
+  if (!identifier) {
+    return data;
+  }
+
+  data.forEach((def) => {
+    let index = -1;
+    const transform = def.transform || [];
+    const aggr = transform.some((t, i) => (index = i, t.type === 'aggregate'));
```

Comment:
This seems simpler. Identifier transforms are only needed to support default selection definitions. Doing it at parse time would require building out all the necessary dataflow architecture to support it, whereas this encapsulates that need in a single function.